c++/src/MemoryPool.cc

```diff
@@ -88,7 +88,7 @@ namespace orc {
 
   template <class T>
   void DataBuffer<T>::reserve(uint64_t newCapacity){
-    if (newCapacity > currentCapacity) {
+    if (newCapacity > currentCapacity || !buf) {
```

Comment:
No, coverity was complaining about needing null checks on buf in other methods. With this change, even if newCapacity is 0, buf is always non-null.